macros/src/derives/format/codegen.rs

```diff
@@ -39,19 +44,16 @@ pub(crate) struct Generics<'a> {
 }
 
 impl<'a> Generics<'a> {
-    pub(crate) fn codegen(generics: &'a mut syn::Generics) -> Self {
+    pub(crate) fn codegen(
+        generics: &'a mut syn::Generics,
+        where_predicates: Vec<WherePredicate>,
+    ) -> Self {
         let mut where_clause = generics.make_where_clause().clone();
         let (impl_generics, type_generics, _) = generics.split_for_impl();
 
-        // Extend where-clause with `Format` bounds for type parameters.
-        for param in &generics.params {
-            if let GenericParam::Type(ty) = param {
-                let ident = &ty.ident;
-
-                where_clause
-                    .predicates
-                    .push(parse_quote!(#ident: defmt::Format));
-            }
+        // Extend where-clause with `Format` bounds for all field types.
+        for predicate in where_predicates {
+            where_clause.predicates.push(predicate);
```

Comment:
You can use `.extend`: https://doc.rust-lang.org/std/iter/trait.Extend.html#tymethod.extend

Reply:
Done